synapse/lib/stormtypes.py

```diff
@@ -731,12 +731,12 @@ async def _libDmonLog(self, iden):
     async def _libDmonAdd(self, text, name='noname', ddef=None):
         text = await tostr(text)
         ddef = await toprim(ddef)
-        varz = await toprim(self.runt.vars)
 
         viewiden = self.runt.snap.view.iden
         self.runt.confirm(('dmon', 'add'), gateiden=viewiden)
 
         # closure style capture of runtime
+        varz = await toprim(self.runt.getScopeVars())
```

Comment:
I'm proposing this fix since capturing vars on the storm query does not work if raw text is passed in